docs/guides/stETH & wstETH Integration Paper.md

```diff
@@ -0,0 +1,396 @@
+
+- TotalShare does not change from slashing, it changes from deposits, withdrawals and burn. As a result, withdrawals and burn share/totalShare may increase.
+
+- The denominator of the fraction share/totalShare is calculated as the sum of all shares, so it cannot be less than the numerator. The ratio can be equal to one in the theoretical case if the address is the only one single holder of stETH.
+
+
+## Contract Logic Summary
+
+- [wstETH](https://docs.lido.fi/contracts/wsteth) is a permissionless constant-balance wrapper token for the rebasing [stETH](https://docs.lido.fi/contracts/lido), the main Ethereum liquid staking token by Lido.
+
+- The amount of stETH an address holds corresponds to the amount of staked ETH plus the Beacon chain rewards accrued and/or penalties inflicted. After withdrawals are enabled in the Merge network, one would be able to convert stETH to ETH 1:1 using the Lido protocol.
+
+- To reflect Beacon chain rewards and/or penalties, balances of all stETH holders are adjusted daily according to the Beacon validators state reported by a quorum of Lido Oracles. In contrast, wstETH token balances remain unchanged: instead, the amount of stETH corresponding to one wstETH changes. Anyone can convert stETH to wstETH and vice versa on-chain through [`wstETH.wrap`](https://docs.lido.fi/contracts/wsteth#wrap) and [`wstETH.unwrap`](https://docs.lido.fi/contracts/wsteth#unwrap) functions.
+
+## Administrative Addresses
+
+No administrative access is built into wstETH.
+
+Note that stETH contract has admin functions that could affect wstETH market value:
+
+- stETH is an upgradable contract, with the upgrade requiring the a DAO vote performed among LDO holders. Voting app is: https://etherscan.io/address/0x2e59a20f205bb85a89c53f1936454680651e618e.
+- The Lido protocol can be stopped by a DAO vote performed among LDO holders. This won't affect wstETH contract in any way except wrapping stETH to wstETH and unwrapping wstETH to stETH won't be possible. Voting app is: https://etherscan.io/address/0x2e59a20f205bb85a89c53f1936454680651e618e. 
+- stETH balances are adjusted upon Lido Oracle reports. This operation doesn't affect wstETH balances in any way. Oracle contract is: https://etherscan.io/address/0x442af784A788A5bd6F42A01Ebe9F287a871243fb
+
+- A wrap followed by an unwrap may result in a lower stETH compared to the initial stETH if the amount of ether locked in the protocol has decreased since the wrap (at this point this may be due to slashing or staking penalties).
+
+## Inheritance Structure
+
+The token contract inherits from the OpenZeppelin's `ERC20Permit`.
+
+## Other
+
+wstETH implements [EIP-2612 Permit](https://eips.ethereum.org/EIPS/eip-2612) standard for `secp256k1`-signed approvals.
+
+
+## Contracts Description Table
```

Comment:
I think we can safely remove this section, tbh.

Reply:
+ Fixed